src/Protocol/ActivityPub/Receiver.php

```diff
@@ -1896,8 +1896,14 @@ private static function processObject(array $object, string $actor)
 		}
 
 		$object_data['unlisted'] = in_array(-1, $object_data['receiver']);
-		unset($object_data['receiver'][-1]);
-		unset($object_data['reception_type'][-1]);
+
+		if (array_key_exists(-1, $object_data['receiver'])) {
+			unset($object_data['receiver'][-1]);
```

Comment:
What's the rationale behind this? I thought `unset` was permissive when it comes to undefined symbols including no-existent array keys?

Reply:
It makes us more aware of the array values. Yes, `unset()` is permissive for undefined symbols but we should move to more specific types. Strict types in arrays will even let PHPStan recognize in higher levels when the call to `unset()` is unnecessary.